### PR TITLE
Added CGNSReader in Reader Classes

### LINKED ISSUE #2410: CGNSReader not showing up in the documentation
The recently added CGNSReader isn't showing up in our documentation at https://docs.pyvista.org/api/readers/_autosummary/pyvista.get_reader.html

image

### PR DESCRIPTION
**Pull request by Kirito-Excalibur**
It was marked as an issue.

### Overview

Added CGNSReader in Reader Classes 

resolves #2410

### CODE REVIEW

**Review by tkoyama010 — APPROVED** [review]
LGTM. Thank you for your contribution!

File: doc/api/readers/index.rst [hunk 1]
```diff
@@ -23,6 +23,7 @@ Reader Classes
    :toctree: _autosummary
    :template: custom-class-template.rst
 
+
```
Comment (MatthewFlamm) [inline doc/api/readers/index.rst:26]: Minor comment: A newline was added here.  Can we instead get rid of the whitespace on the adjacent empty line?

Comment (MatthewFlamm) [reply doc/api/readers/index.rst:26]: GitHub is weird about showing whitespace diff, so I'm not sure if I've got this right.

Comment (Kirito-Excalibur) [reply doc/api/readers/index.rst:26]: Line 26?

Comment (MatthewFlamm) [reply doc/api/readers/index.rst:26]: Line 25 looks to have spaces to me in some views on GitHub.  If we remove those, do we need the added newline in this PR?

Comment (Kirito-Excalibur) [reply doc/api/readers/index.rst:26]: No not really. Btw from what i can see. Line 25 is empty. Full blank

Comment (MatthewFlamm) [inline doc/api/readers/index.rst:26]: Sorry for the confusion, the added line had the extra whitespace.  Unless this is needed somehow, let's remove that.

```suggestion
```

Comment (Kirito-Excalibur) [reply doc/api/readers/index.rst:26]: okay. btw where are mods. One approval is left for commit.

**Review by akaszynski — APPROVED** [review]
> okay. btw where are mods. One approval is left for merging.

Mods are here.

**Pull request merged**